src/mongo/db/storage/csv_file.cpp

```diff
@@ -0,0 +1,338 @@
+/**
+ *    Copyright (C) 2024-present Codein, Inc.
+ *
+ *    This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the Server Side Public License, version 1,
+ *    as published by MongoDB, Inc.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    Server Side Public License for more details.
+ *
+ *    You should have received a copy of the Server Side Public License
+ *    along with this program. If not, see
+ *    <http://www.mongodb.com/licensing/server-side-public-license>.
+ *
+ *    As a special exception, the copyright holders give permission to link the
+ *    code of portions of this program with the OpenSSL library under certain
+ *    conditions as described in each individual source file and distribute
+ *    linked combinations including the program with the OpenSSL library. You
+ *    must comply with the Server Side Public License in all respects for
+ *    all of the code used other than as permitted herein. If you modify file(s)
+ *    with this exception, you may extend this exception to your version of the
+ *    file(s), but you are not obligated to do so. If you do not wish to do so,
+ *    delete this exception statement from your version. If you delete this
+ *    exception statement from all source files in the program, then also delete
+ *    it in the license file.
+ */
+#include <cassert>
+#include <cerrno>
+#include <chrono>
+#include <cstdio>
+#include <fmt/format.h>
+#include <fstream>  // IWYU pragma: keep
+#include <string>
+#include <sys/stat.h>
+
+#include "mongo/base/error_codes.h"
+#include "mongo/base/string_data.h"
+#include "mongo/db/query/query_knobs_gen.h"
+#include "mongo/db/storage/csv_file.h"
+#include "mongo/db/storage/io_error_message.h"
+#include "mongo/util/assert_util.h"
+#include "mongo/util/time_support.h"
+
+namespace mongo {
+using namespace fmt::literals;
+
+CsvFileInput::CsvFileInput(const std::string& fileRelativePath, const std::string& metaFileRelative)
+    : _fileAbsolutePath((externalPipeDir == "" ? kDefaultPipePath : externalPipeDir) +
+                        fileRelativePath),
+      _metadataAbsolutePath((externalPipeDir == "" ? kDefaultPipePath : externalPipeDir) +
+                            metaFileRelative),
+      _ifs() {
+    uassert(200000400,
+            "File path must not include '..' but {} does"_format(_fileAbsolutePath),
+            _fileAbsolutePath.find("..") == std::string::npos);
+}
+
+CsvFileInput::~CsvFileInput() {
+    _ifs.close();
+}
+
+void CsvFileInput::doOpen() {
+    std::string metadata;
+    _ifs.open(_metadataAbsolutePath.c_str(), std::ios::binary | std::ios::in);
```

Comment:
The metadata file is a text file. Why do we open it with `std::ios::binary` flag?